### PR TITLE
JSON/XML/CSV nil options

### PR DESCRIPTION
**Pull request by mgilbir**
This PR enables the generation of JSON and XML when the corresponding option objects are nil.

This is useful when one doesn't care about the structure of the generated JSON/XML as long as it is valid.

Some of the tests check for a well formed `JSONOptions` and fail when it is missing required parameters (eg. `Fields` or `RowCount`). While it may seem that a nil `JSONOptions` would be equally malformed, one can argue that when the options are nil it means that they are *not* specified instead of misspecified.

In that case, we can leverage the gofakeit mechanics to generate a random `JSONOptions`/`XMLOptions`, and proceed with the generation as usual.

Tests pass and code coverage is unchanged.

### CODE REVIEW

**Comment by brianvoe** [comment]
I think this looks great! nice job!

The only think i question is setting firstname or autoincrement on the Field. I almost think that one should be required.

Let me know your thoughts.

**Comment by mgilbir** [comment]
Ah, now I see what you mean. For my use case, as long as the data is valid
I was happy.

I'll think if there is something easy that can be done 🤔

**Comment by brianvoe** [comment]
Ya from a programmers perspective I agree. as long as my code sees it as good then im good. But I do have a decent amount of people that use it for visuals. Things like front end. I am also building gofakeit.com and would like it if this feature had more visually appealing outputs as well.

**Comment by mgilbir** [comment]
Would an output like this be "visual" enough?

```
{
    "property_1": "Mary",
    "property_2": 42,
    ...
}
```

or do you want to go all the way to named properties like:

```
{
    "name": "Mary",
    "age": 42,
    ...
}
```

**Comment by brianvoe** [comment]
ya that looks good. I would think randomly between 2 and 8 fields and maybe picking between the top 8 most used ones.

**Comment by mgilbir** [comment]
I tried something different. 

It now picks a random number of existing parameterless FuncLookup and uses that to generate data.

An example looks like this:
```
[
  {
    "loglevel": "trace",
    "minecraftweapon": "shield",
    "minecraftbiome": "ocean",
    "streetprefix": "South",
    "minute": 40,
    "uint16": 47109,
    "cartransmissiontype": "Automatic",
    "language": "Portuguese",
    "dinner": "Ww shrimp scampi",
    "errorhttpserver": {}
  },
  {
    "loglevel": "error",
    "minecraftweapon": "sword",
    "minecraftbiome": "ice spike",
    "streetprefix": "Port",
    "minute": 24,
    "uint16": 7319,
    "cartransmissiontype": "Manual",
    "language": "Oromo",
    "dinner": "Blackened tuna bites with cajun mustard",
    "errorhttpserver": {}
  }
]
```

or 

```
[
  {
    "dinner": "Cafe mocha latte",
    "hobby": "Shoemaking",
    "gamertag": "DullMacaw151",
    "adverb": "hardly",
    "uint32": 2581418833,
    "emojidescription": "black square button",
    "beerstyle": "Fruit Beer",
    "emojitag": "france",
    "pronounpersonal": "we",
    "bird": "ibis"
  },
  {
    "dinner": "Garlic prime rib",
    "hobby": "Bowling",
    "gamertag": "ConcerningPug",
    "adverb": "recently",
    "uint32": 2058102316,
    "emojidescription": "tangerine",
    "beerstyle": "Vegetable Beer",
    "emojitag": "lick",
    "pronounpersonal": "I",
    "bird": "duck"
  }
]
```

**Comment by brianvoe** [comment]
I think this looks great! The only thing i would like to figure out is being able to get random lookup functions to run without needing to add internalFuncLookups. The lookup funcs are used in cmd, server and gofakeit.com. So adding to it would have it show up on the list of available functions to run.

**Comment by mgilbir** [comment]
I'm not sure I'm following. In particular I'm confused about the interest/value of having this particular function show up on the list of available functions to run from cmd, server and gofakeit.com.

The lookup function I added 'internal_ExampleFields' is returning a `gofakeit.Field` which I think limits the value of exposing it to end users automatically.

That is the reason why I added the `internalFuncLookups` so that it is possible to have a map with functions that are meant to be only used by gofakeit itself without exposing them to end users. 

The reason to return a `gofakeit.Field` is because that is the value that we are trying to fake in the struct tag of JSONOptions/XMLOptions/CSVOptions:
https://github.com/brianvoe/gofakeit/pull/231/commits/49efc36a2fcfd7c2eb371908a74b756306686449#diff-e848e95f31309994dbd646cda14ba74ae0239fba72b388bb9cc0928057c50980R14

To sum up my understanding, and I may be missing something obvious, I see three ways of faking that value:
- Expose a custom FuncLookup returning a `gofakeit.Field` to end users (IMO confusing unless they have a use for `gofakeit.Field`)
- Have a custom FuncLookup returning a `gofakeit.Field` not exposed to end users (current implementation)
- Use the mechanism in the other pull request (#230) to implement the `gofakeit.Fakeable` interface for `gofakeit.Field` (solves it without needing to introduce the `internalFuncLookup`)

What am I missing? 🤔 

I hope I'm not being too obnoxious. I really appreciate you engaging and the project you've built! 😄

**Comment by brianvoe** [comment]
Merged! Thanks for your contributions. I think people will enjoy the addition.
